### PR TITLE
Provide a per-week breakdown of performance.

### PR DESCRIPTION
**Pull request by james-w**
Provide some basic per-week stats from the source data.

We map the timestamps to a week number (including year),
then group by that value. We provide a build count and
the success percentage for each week.

I also made a small change to use DataFrame.assign to create
the test_case_name column, as I found the API while getting
warnings about using assignment for the new columns. This
means that all locations do this consistently.

### CODE REVIEW

File: jenkins/_jenkins.py [hunk 1]
```diff
@@ -171,9 +174,74 @@ def classify_build_log(log, path):
...
+
+    :param str result_str: test result status
+        ("SUCCESS", "FAILED", etc.)
+    :return int: 100 if the test passed, 0 if it
+        failed. These mean of these values will
+        be the percentage of successful tests.
+    """
+    if result_str == SUCCESS:
+        return 100
+    return 0
```
Comment (jml) [inline jenkins/_jenkins.py:205]: :question: Maybe a bit more sciency to use `1` for success and then multiply out at the end.

Comment (james-w) [inline jenkins/_jenkins.py:205]: Indeed, but the search results for how to display a float as a percentage
showed how to override the global pandas float formatting. I guess an
extra map before display would probably work.

File: jenkins/_jenkins.py [hunk 2]
```diff
@@ -171,9 +174,74 @@ def classify_build_log(log, path):
...
+    df = pandas.DataFrame(builds).assign(
+        week_number=make_week_numbers,
+        numeric_result=make_numeric_results,
+    )
+    print df.groupby('week_number').agg(
+        {'numeric_result': {
+            'test runs': lambda x: x.count(),
+            'success percentage': numpy.mean
+        }}
+    )['numeric_result'].head()
```
Comment (jml) [inline jenkins/_jenkins.py:244]: :question: Probably not worth spending the time on this (unless you want to learn more pandas), but I'm pretty sure you can do this with it's [built-in resampling support](http://pandas.pydata.org/pandas-docs/version/0.17.1/timeseries.html#resampling)

File: jenkins/_jenkins.py [hunk 3]
```diff
@@ -171,9 +174,74 @@ def classify_build_log(log, path):
     return "Unknown"
 
 
+def get_week_number(timestamp):
+    """
+    Return a week number for the timestamp.
+
+    This combines the year and the week number so that
+    it can handle per-week processing of data that
+    spans multiple years.
```
Comment (jml) [inline jenkins/_jenkins.py:183]: :question: FWIW, I think our coding standard allows wider wrapping here.

**Comment by jml** [comment]
Great, thanks. Only optional comments. Merge when ready.

**Pull request merged**